tests/common/test_event.py

```diff
@@ -163,6 +165,23 @@ def test_save_event(self):
         add_event('test', message='test event')
         self.assertTrue(len(os.listdir(self.tmp_dir)) == 2)
 
+    def test_save_event_message_with_non_ascii_characters(self):
+        stdout = open(os.path.join(data_dir, "events", "collect_and_send_extension_stdout_stderror",
+                                   "dummy_stdout_with_non_ascii_characters"), mode="r+b")
+        stderr = open(os.path.join(data_dir, "events", "collect_and_send_extension_stdout_stderror",
+                                   "dummy_stderr_with_non_ascii_characters"), mode="r+b")
+
+        msg = read_output(stdout, stderr)
+        duration = elapsed_milliseconds(datetime.utcnow())
+        log_msg = "{0}\n{1}".format("DummyCmd", "\n".join([line for line in msg.split('\n') if line != ""]))
+
+        add_event('test_extension', message=log_msg, duration=duration)
+        for tld_file in os.listdir(self.tmp_dir):
+            event_str = MonitorHandler.collect_event(os.path.join(self.tmp_dir, tld_file))
```

Comment:
can you fix the implementation of collect_event to cleanup (delete) the file if it cannot decode it?

```
 def collect_event(evt_file_name):
        try:
            logger.verbose("Found event file: {0}".format(evt_file_name))
            with open(evt_file_name, "rb") as evt_file:
                # if fail to open or delete the file, throw exception
                data_str = evt_file.read().decode("utf-8", 'ignore')
            logger.verbose("Processed event file: {0}".format(evt_file_name))
            os.remove(evt_file_name)
            return data_str
        except IOError as e:
            msg = "Failed to process {0}, {1}".format(evt_file_name, e)
            raise EventError(msg)
```